### PR TITLE
Update regenerator to newest version

### LINKED ISSUE #6: Update regenerator
When browserifying pac-resolver, I've been getting an issue with `fs` in `regenerator`:

```js
  var runtime = options.includeRuntime ? fs.readFileSync(
    regenerator.runtime.dev, "utf-8"
  ) + "\n" : "";
``` 

This throws an error, where `fs` is not defined due to the path being a variable in fs.readFileSync (this is a known issue: for instance, here https://github.com/substack/node-browserify/issues/883). 

The newest version of `regenerator` doesn't have this error.

### PR DESCRIPTION
**Pull request by RichardLitt**
I've updated the package in `package.json` and fixed the code to fit the new exports. Closes #6

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ function generate (str, opts) {
 
   // use `facebook/regenerator` for node < v0.11 support
   // TODO: don't use regenerator if native generators are supported...
-  js = regenerator(js, { includeRuntime: false });
+  js = regenerator.cpmpile(js, { includeRuntime: false }).code;
```
Comment (TooTallNate) [inline index.js]: typo?

**Comment by TooTallNate** [comment]
Travis saying `wrapGenerator not defined` now: https://travis-ci.org/TooTallNate/node-pac-resolver/jobs/51549982

**Comment by RichardLitt** [comment]
Hmm. Fixed typo, looking into wrapGenerator.

**Comment by TooTallNate** [comment]
I think it's `regeneratorRuntime.wrap()` now.

**Comment by RichardLitt** [comment]
It is. Changed the commit again, should be good now.

File: index.js [hunk 2]
```diff
@@ -32,8 +32,8 @@ var weekdayRange = require('./weekdayRange');
 
 module.exports = generate;
 
-// cache the `facebook/regenerator` wrapGenerator function.
-var wg = vm.runInNewContext(regenerator('', { includeRuntime: true }) + ';wrapGenerator');
+// cache the `facebook/regenerator` wrap function for the Generator object.
+var wg = vm.runInNewContext(regenerator.compile('', { includeRuntime: true }).code + ';regeneratorRuntime.wrap()');
```
Comment (TooTallNate) [inline index.js]: Don't invoke the function! We want a reference to the function itself, not its return value!

File: index.js [hunk 3]
```diff
@@ -90,7 +90,7 @@ function generate (str, opts) {
 
   // use `facebook/regenerator` for node < v0.11 support
   // TODO: don't use regenerator if native generators are supported...
-  js = regenerator(js, { includeRuntime: false });
+  js = regenerator.compile(js, { includeRuntime: false }).code;
 
   // filename of the pac file for the vm
   var filename = opts && opts.filename ? opts.filename : 'proxy.pac';
```
Comment (TooTallNate) [inline index.js]: Also, in the part where we're setting `wrapGenerator = wg` on the sandbox object, we'll need to make that be a `regeneratorRuntime: { wrap: wg }` object.

Comment (TooTallNate) [inline index.js]: At that point, we might as well just pluck out the entire `regeneratorRuntime` object, in case it's used for other stuff.

Comment (RichardLitt) [inline index.js]: Something like this `  sandbox.regeneratorRuntime = wg;` ?

**Comment by RichardLitt** [comment]
How's that look?

**Comment by TooTallNate** [comment]
I _think_ that looks good! Let's wait to see what Travis has to say :)

**Comment by RichardLitt** [comment]
Hmm. `npm test` locally works for me, and I'm not sure what that error is in Travis. Looks like `npm install` failed for some reason?

**Comment by TooTallNate** [comment]
That's just old npm from v0.8 complaining about packages using the unsupported `^` semver syntax. I've dropped v0.8 from the tests now.

**Comment by TooTallNate** [comment]
Thanks, merged!

**Comment by RichardLitt** [comment]
Awesome. Thanks!

**Pull request merged**